inspector/src/components/actionTabs/tabs/propertyGrids/materials/textures/propertiesBar.tsx

```diff
@@ -1,144 +1,149 @@
-                    </div>
-                </>}
-                <div className='tab' id='right-tab'>
-                    <div className='content'>
-                        <img title='Reset' className='icon button' src={resetButton} onClick={() => this.props.resetTexture()}/>
-                        <label>
-                            <input
-                                accept='.jpg, .png, .tga, .dds, .env'
-                                type='file'
-                                onChange={
-                                    (evt : React.ChangeEvent<HTMLInputElement>) => {
-                                        const files = evt.target.files;
-                                        if (files && files.length) {
-                                            this.props.uploadTexture(files[0]);
-                                        }
-                                
-                                        evt.target.value = "";
-                                    }
-                                }
-                            />
-                            <img
-                                title='Upload'
-                                className='icon button'
-                                src={uploadButton}
-                            />
-                        </label>
-                        <img title='Save' className='icon button' src={saveButton} onClick={() => this.props.saveTexture()}/>
-                    </div>
-                </div>
-        </div>;
-    }
+import * as React from 'react';
+import { BaseTexture } from 'babylonjs/Materials/Textures/baseTexture';
+import { PixelData } from './textureCanvasManager';
+
+interface PropertiesBarProps {
+    texture: BaseTexture;
+    saveTexture(): void;
+    pixelData: PixelData;
+    face: number;
+    setFace(face : number): void;
+    resetTexture() : void;
+    resizeTexture(width: number, height: number) : void;
+    uploadTexture(file : File) : void;
+}
+
+interface PropertiesBarState {
+    width: number;
+    height: number;
+}
+
+interface PixelDataProps {
+    name : string;
+    data?: number;
+}
+
+const resetButton = require('./assets/reset.svg');
+const uploadButton = require('./assets/upload.svg');
+const saveButton = require('./assets/save.svg');
+const babylonLogo = require('./assets/babylonLogo.svg');
+
+const posX = require('./assets/posX.svg');
+const posY = require('./assets/posY.svg');
+const posZ = require('./assets/posZ.svg');
+const negX = require('./assets/negX.svg');
+const negY = require('./assets/negY.svg');
+const negZ = require('./assets/negZ.svg');
+
+const resizeButton = require('./assets/resizeTool.svg');
+
+const mipUp = require('./assets/mipUp.svg');
+const mipDown = require('./assets/mipDown.svg');
+
+const faces = [
+    posX,
+    negX,
+    posY,
+    negY,
+    posZ,
+    negZ
+]
+
+function PixelData(props : PixelDataProps) {
```

Comment:
should start with _ if private (not exported)

Reply:
And I'm not a big fan of "flying" functions, so maybe just move them as private functions inside the propertiesBar?

Reply:
So the PixelData component should instead be _pixelData and should be a private function of the class?

And faces/the icons should be private member variables?

Reply:
@deltakosh I moved everything inside the class, does that look better?